### PR TITLE
Dev App Server v1 integration

### PR DESCRIPTION
**Pull request by loosebazooka**
- Services replaces AppYamls (setting AppYamls reveals deprecation warning)
- By default, we use dev appserver v1
- users can swap between v1 and v2 by setting
```
appengine {
  run {
    serverVersion = "1" or "2-alpha"
  }
}
```

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/gradle/appengine/standard/task/DevAppServerHelper.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  static final List<String> SERVER_VERSIONS = ImmutableList.of(V1, V2);
+
+  private Validator validator = new Validator();
+
+  public AppEngineDevServer getAppServer(CloudSdk sdk, Run run) {
+
+    String serverVersion = run.getServerVersion();
+    validator.validateServerVersion(serverVersion);
+
+    if (V1.equals(serverVersion)) {
```
Comment (akerekes) [inline src/main/java/com/google/cloud/tools/gradle/appengine/standard/task/DevAppServerHelper.java]: You could use `switch` instead

File: src/main/java/com/google/cloud/tools/gradle/appengine/standard/task/DevAppServerHelper.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  public StopConfiguration getStopConfiguration(Run run) {
+
+    String serverVersion = run.getServerVersion();
+    validator.validateServerVersion(serverVersion);
+
+    DefaultStopConfiguration stop = new DefaultStopConfiguration();
+
+    if (V1.equals(serverVersion)) {
```
Comment (akerekes) [inline src/main/java/com/google/cloud/tools/gradle/appengine/standard/task/DevAppServerHelper.java]: You could use `switch` instead

File: src/main/java/com/google/cloud/tools/gradle/appengine/standard/task/DevAppServerHelper.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  @VisibleForTesting
+  static class Validator {
+
+    @VisibleForTesting
+    void validateServerVersion(String serverVersion) throws ProjectConfigurationException {
+      if (!SERVER_VERSIONS.contains(serverVersion)) {
+        throw new ProjectConfigurationException(
+            "Invalid serverVersion '" + serverVersion + "' please use " + SERVER_VERSIONS,
```
Comment (akerekes) [inline src/main/java/com/google/cloud/tools/gradle/appengine/standard/task/DevAppServerHelper.java]: `use one of`

File: src/test/java/com/google/cloud/tools/gradle/appengine/standard/AppEngineStandardPluginTest.java [hunk 4]
```diff
@@ -76,94 +76,14 @@ public void testDeploy_taskTree() throws IOException {
         .withArguments("appengineDeploy", "--dry-run")
         .build();
 
-    final List<String> expected = ImmutableList
-        .of(":compileJava", ":processResources", ":classes", ":war", ":explodeWar", ":assemble",
+    final List<String> expected = Arrays
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/gradle/appengine/standard/AppEngineStandardPluginTest.java]: Why the change?

Comment (loosebazooka) [inline src/test/java/com/google/cloud/tools/gradle/appengine/standard/AppEngineStandardPluginTest.java]: I dont' know? I'll change this back

File: src/test/java/com/google/cloud/tools/gradle/appengine/standard/extension/RunExtensionTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+
+public class RunExtensionTest {
+
+  @Rule
+  public TemporaryFolder tmpDir = new TemporaryFolder();
+
+  @Rule
+  public ExpectedException exception = ExpectedException.none();
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/gradle/appengine/standard/extension/RunExtensionTest.java]: If you don't expect exceptions, they you don't need this field.

File: src/main/java/com/google/cloud/tools/gradle/appengine/standard/extension/Run.java [hunk 6]
```diff
@@ -74,18 +76,18 @@ public void setStartSuccessTimeout(int startSuccessTimeout) {
...
+  public void setServerVersion(String serverVersion) throws ProjectConfigurationException {
+    this.serverVersion = serverVersion;
   }
 
-  @Override
-  public List<File> getServices() {
-    return null;
+  @Deprecated
+  public void setAppYamls(Object appYamls) {
+    project.getLogger().warn("'appYamls' is deprecated, this parameter will set 'services'. Please use 'services' in the future.");
```
Comment (akerekes) [inline src/main/java/com/google/cloud/tools/gradle/appengine/standard/extension/Run.java]: We can leave out `please`.
